internal/faultinjectors/mirroring_test.go

```diff
@@ -167,3 +337,65 @@ func oopsAllAttachFrameNames(allFrames []*frames.Frame) []string {
 
 	return names
 }
+
+// LoadStateMap loads up some pre-recorded ATTACH frames int our statemap:
+// - a receiver, with local channel/handle: 200/200, remote 0/0
+// - a sender, with local channel/handle 300/300, remote 1001/1002
+func loadStateMap(t *testing.T) *proto.StateMap {
+	sm := proto.NewStateMap()
+
+	// handle/channel are 200 for these example files.
+	reader, err := os.Open("testdata/receiver_attach_frames.json")
+	require.NoError(t, err)
+	defer reader.Close()
+	decoder := json.NewDecoder(reader)
+
+	var line *logging.JSONLine
+	require.NoError(t, decoder.Decode(&line))
+	require.NotEmpty(t, line)
+
+	sm.AddFrame(line.Direction == logging.DirectionOut, line.Frame)
+
+	line = nil
+	require.NoError(t, decoder.Decode(&line))
+	require.NotEmpty(t, line)
+
+	sm.AddFrame(line.Direction == logging.DirectionOut, line.Frame)
+
+	// handle/channel are 300 for these example files.
+	reader, err = os.Open("testdata/sender_attach_frames.json")
+	require.NoError(t, err)
+	defer reader.Close()
+	decoder = json.NewDecoder(reader)
+
+	line = nil
+	require.NoError(t, decoder.Decode(&line))
+	require.NotEmpty(t, line)
+
+	sm.AddFrame(line.Direction == logging.DirectionOut, line.Frame)
+
+	line = nil
+	require.NoError(t, decoder.Decode(&line))
+	require.NotEmpty(t, line)
+
+	sm.AddFrame(line.Direction == logging.DirectionOut, line.Frame)
+
+	// sanity check, all the frames are loaded and with the proper direction.
+	out := true
+	localSenderAF := sm.LookupAttachFrame(out, 300, 300)
+	require.Equal(t, "testQueue", localSenderAF.Body.Address(out))
+
+	out = false
+	remoteSenderAF := sm.LookupAttachFrame(out, 1001, 1002)
+	require.Equal(t, "testQueue", remoteSenderAF.Body.Address(out))
+
+	out = true
+	localReceiveAF := sm.LookupAttachFrame(out, 200, 200)
+	require.Equal(t, "testQueue", localReceiveAF.Body.Address(out))
+
+	out = false
+	remoteReceiverAF := sm.LookupAttachFrame(out, 0, 0)
+	require.Equal(t, "testQueue", remoteReceiverAF.Body.Address(out))
+
```

Comment:
would we also want to add cbs link tests? Or not, since they're just a sender/receiver pair?

Reply:
Added an example for that too - it's basically the same, but it's still nice to see.